### PR TITLE
Edu benefits form bug fixes

### LINKED ISSUE #3300: Date field in benefit relinquishment is breaking form
That's a bit dramatic, but it seems the dat range for that field doesn't include future dates, but is not displaying an error message. The result is that user can't move to the next screen if they enter in a future date.

e.g. 10/6 worked, 1/1/2017 didn't.

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #3300, #3288, #3272, #3254, #3250, #3170

This is a grab bag of bug fixes, as well as some style updates to make the review page match the button styles and colors in other parts of the form.

Cleaned up ROTC styles when fixing add another bug:
screen shot 2016-10-06 at 7 17 32 pm

Removed note when in review panel:
screen shot 2016-10-06 at 7 18 21 pm

Updated review button color:
screen shot 2016-10-07 at 9 21 18 am

Update benefit selection review (button colors are prior to above change):
screen shot 2016-10-06 at 7 19 33 pm

Don't show 'No' when user didn't make a choice
screen shot 2016-10-06 at 7 20 26 pm

Update normal button color:
screen shot 2016-10-07 at 9 25 12 am

### CODE REVIEW

File: src/sass/hca.scss [hunk 1]
```diff
@@ -202,3 +202,35 @@ button.hca-button-disabled {
     text-align: left;
   }
 }
+
```
Comment (jbalboni) [inline src/sass/hca.scss]: Moved these styles to HCA so it doesn't break.

File: src/sass/modules/_m-button.scss [hunk 2]
```diff
@@ -7,12 +7,12 @@ button,
 [type="submit"],
 [type="reset"],
 [type="image"] {
-  background-color: $color-primary-darker;
```
Comment (jbalboni) [inline src/sass/modules/_m-button.scss]: FYI: making those button style changes here.

Comment (bshyong) [inline src/sass/modules/_m-button.scss]: LGTM; thanks for the heads up!

File: src/js/edu-benefits/utils/validations.js [hunk 3]
```diff
@@ -303,8 +303,8 @@ function isValidBenefitsHistoryPage(data) {
 }
 
 function isValidRotcScholarshipAmount(data) {
-  return isValidField(isValidMonetaryValue, data.amount)
-    && isValidField(isValidYear, data.year);
+  return (isBlank(data.amount.value) || isValidField(isValidMonetaryValue, data.amount))
+    && (isBlank(data.amount.value) || isValidField(isValidYear, data.year));
```
Comment (akainic) [inline src/js/edu-benefits/utils/validations.js]: Did you mean `isBlank(data.year.value)` here? Also, does it make sense for us to allow additional entries to be added if the values are blank?

File: src/sass/modules/_m-button.scss (hunk 2 above)
Comment (akainic) [inline src/sass/modules/_m-button.scss]: This button color looks so much better :)

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionReview.jsx [hunk 4]
```diff
@@ -9,8 +9,8 @@ export default class BenefitsSelectionReview extends React.Component {
       <table className="review usa-table-borderless">
         <tbody>
           <tr>
-            <td>Post-9/11 GI Bill (Chapter 33)</td>
-            <td>{this.props.data.chapter33 ? 'Yes' : 'No'}</td>
+            <td>Benefits selection:</td>
```
Comment (akainic) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionReview.jsx]: When testing this locally, the relinquishment date isn't appearing in the Review section (I entered a future date).

**Review by akainic — COMMENTED** [review]
Couple of minor notes, but looks good!

File: src/js/edu-benefits/utils/validations.js (hunk 3 above)
Comment (jbalboni) [inline src/js/edu-benefits/utils/validations.js]: Yep, I'll fix that.

For the second question, I don't think we have a good way of preventing that. It makes the growable table even more complicated. It's something we should probably talk about for the future, though.

File: src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionReview.jsx (hunk 4 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/benefits-eligibility/BenefitsSelectionReview.jsx]: Good catch. It's tripped me up several times that `isValidDate` only allows dates in the past.

**Pull request merged**